scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt

```diff
@@ -236,6 +241,7 @@ class ContainerFactory(
         throw IllegalStateException("You cannot start a workspace with no solutionId defined")
     val solution =
         solutionService.findSolutionById(organizationId, workspace.solution.solutionId ?: "")
+    azureContainerRegistryClient.checkSolutionImage(solution.repository!!, solution.version!!)
```

Comment:
Well, you can keep this check on solution image here.
Plus, could you add this check also when updateSolution endpoint is called (before the save obviously) in order to verify that, when solution is updtaing, targeted container exists